### ISSUE TITLE
Missing data issue #129 and MaybeNumber

### ISSUE THREAD

**Issue by owattenmaker** [FormidableLabs/victory-native-xl#135]
Since the recent update (40.0.0) I've noticed that #129 works for `undefined` values, but for data that is `null` the output will still populate as a value of 0.

This is due to the `asNumber` function. I know #129 removed some of the calls to this, but it didn't fix the root cause of the issue being the following:
```ts
asNumber(null)
>>> 0
asNumber(undefined)
>>> NaN
```

One fix for this is to update the `asNumber` function like so:
```ts
export const asNumber = (val: unknown): number => {
  "worklet";
  if (val === null) return NaN
  return typeof val === "number" ? val : Number(val);
};
```

**Comment by gksander** [comment]
Thanks for reporting this in – I'll take a look!

**Comment by gksander** [comment]
#138 was merged, release as 40.0.2, and I _think_ that should resolve the issue you're seeing around `null` values. If not, let's reopen this!

**Comment by owattenmaker** [comment]
Thank you!
